src/Services/Virtualization/HttpHeaderTools.cs

```diff
@@ -166,6 +166,12 @@ public static void SetCacheControlHeaders(int cacheForSeconds)
         public static void SetCacheControlHeaders(int cacheForSeconds, HttpCacheability httpCacheability)
         {

```

Comment:
This method is too often accesses a member within the same chain. It would be better to pin the target member into a local variable. For example: `var cache = HttpContext.Current.Response.Cache`. And use the appropriate methods of the pinned variable: `cache.SetNoStore()` etc.